doc/dev/developer_guide/basic-workflow.rst

```diff
@@ -290,11 +317,16 @@ will need to force push your branch with:
 
    git push --force origin fix_1
 
+Why do we take these extra steps instead of simply adding additional commits
+the the PR?  It is best practice for a PR to consist of a single commit; this
```

Comment:
Well, actually no. In case it is possible to do so, we encourage contributors to split PRs into multiple smaller, independent, commits. 

See https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#separate-your-changes